system/Validation/Validation.php

```diff
@@ -559,12 +559,14 @@ protected function loadRuleSets()
      * same format used with setRules(). Additionally, check
      * for {group}_errors for an array of custom error messages.
      *
-     * @return array|ValidationException|null
+     * @throws ValidationException
+     *
+     * @return array
      */
     public function loadRuleGroup(?string $group = null)
     {
         if (empty($group)) {
-            return null;
```

Comment:
What did "null" mean? Would anyone have been using this? Why do we even allow `null` as a parameter if we're just going to return immediately...

Reply:
It came from https://github.com/codeigniter4/CodeIgniter4/pull/1769/files#diff-fa78800b3fc00024660b9f8090bd6bee512cd4fbedca371233eb5faa9c57cb06

I'm not sure. It seems just a mistake.

Reply:
The return type was void at first, and changed to void and array.
https://github.com/codeigniter4/CodeIgniter4/commit/37dbc1a8#diff-fa78800b3fc00024660b9f8090bd6bee512cd4fbedca371233eb5faa9c57cb06

But `$this->rules` is an array, so it should be return `[]` when there are no rules.

Reply:
> Why do we even allow null as a parameter if we're just going to return immediately...

At least, now there is no place that expects null in the system code.
`getValidationRules()` uses this method and must return an array.
So if this returns null, it causes an error.

Reply:
I agree with your change. Probably should leave the parameter, makes me a bit sad though. Some days I just want to break everything and redo it the right way and release version 5 this month...